Marketplace/dodgeball/dodgeball.js

```diff
@@ -0,0 +1,288 @@
+        }
+    }
+
+    function proxCheck() {
+        var ballPos = Entities.getEntityProperties(_entityID, ['position']).position;
+        var avatarReleasingBallPosition = AvatarList.getAvatar(releasingAvatarID).position;
+        var ballFromAvatar = Vec3.distance(avatarReleasingBallPosition, ballPos);
+
+        if (ballFromAvatar > 1.0) {
+            var isAnyAvatarInRange = AvatarList.isAvatarInRange(ballPos, 0.80);
+            if (isAnyAvatarInRange) {
+                clearProxCheck();
+                particleExplode();
+            }
+        }
+    }
+
+    this.preload = function (entityID) {
+        _entityID = entityID;
+
+        Entities.editEntity(_entityID, {
+            userData: JSON.stringify({grabbableKey: {grabbable: true}})
+        });
+        BALL_SOUNDS.push(SoundCache.getSound(Script.resolvePath('bounce01.wav')));
+        BALL_SOUNDS.push(SoundCache.getSound(Script.resolvePath('bounce02.wav')));
+        BALL_SOUNDS.push(SoundCache.getSound(Script.resolvePath('bounce03.wav')));
+        BALL_SOUNDS.push(SoundCache.getSound(Script.resolvePath('bounce04.wav')));
+        BALL_SOUNDS.push(SoundCache.getSound(Script.resolvePath('bounce05.wav')));
+        HIT_SOUND = SoundCache.getSound(Script.resolvePath('hit.wav'));
+    };
+
+    this.startDistanceGrab = function (thisEntityID, triggerHandAndAvatarUUIDArray) {
+        clearProxCheck();
+        var triggerHand = triggerHandAndAvatarUUIDArray[0];
+        var avatarUUID = triggerHandAndAvatarUUIDArray[1];
+
+        var ballPos = Entities.getEntityProperties(_entityID, ['position']).position;
+        var MAX_DISTANCE_GRAB = 2; // meter
+
+        if (Vec3.distance(ballPos, AvatarList.getAvatar(avatarUUID).position) > MAX_DISTANCE_GRAB) {
+            Messages.sendMessage(FORCE_DROP_CHANNEL, triggerHand, true);
+        }
+
+    };
+
+    this.startNearGrab = function (thisEntityID, triggerHandAndAvatarUUIDArray) {
+        clearProxCheck();
+    };
+
+    this.releaseGrab = function (thisEntityID, triggerHandAndAvatarUUIDArray) {
+        canMakeBounceNoise = true;
+        releasingAvatarID = triggerHandAndAvatarUUIDArray[0];
+
+        if (particleTrailEntity === null) {
+            particleTrail();
+        }
+
+        Script.setTimeout(function () {
+            proxInterval = Script.setInterval(proxCheck, 50);
+        }, 200); // Setting a delay to give it time to leave initial avatar without proc.
+
+        proxTimeout = Script.setTimeout(function () {
+            clearProxCheck();
+        }, 10000);
+    };
+
+    this.collisionWithEntity = function (thisEntityID, collisionEntityID, collisionInfo) {
+
+        if (canMakeBounceNoise){ // a simple debounce
+            playBounceSound(collisionInfo.contactPoint);
+            canMakeBounceNoise = false;
+            Script.setTimeout(function () {
+                canMakeBounceNoise = true;
+            }, 2000);
+        }
+
+        clearProxCheck();
+    };
+
```

Comment:
forgot to mention this, you'll need a  `this.unload` function opposing the `this.preload` because if no one touches or collides with the entity `clearProxCheck()` might not be called before the entity is destroyed or script restarted.